tests/test_crypto/test_helpers.py

```diff
@@ -55,3 +71,57 @@ def tests_private_keys(self):
                 CUR_PATH, "data", "fet_private_key_wrong.txt"
             )
             _try_validate_ethereum_private_key_path(private_key_path)
+
+    @patch("aea.crypto.helpers.logger")
+    def tests_generate_wealth_fetchai(self, mock_logging):
+        """Test generate wealth for fetchai."""
+        address = "my_address"
+        result = Response()
+        result.status_code = 500
+        with patch.object(requests, "post", return_value=result):
+            _try_generate_testnet_wealth(identifier=FETCHAI, address=address)
+            assert mock_logging.error.called
+
+        result.status_code = 200
+        with pytest.raises(SystemExit):
+            with patch.object(requests, "post", return_value=result):
+                _try_generate_testnet_wealth(identifier=FETCHAI, address=address)
+
+    @patch("aea.crypto.helpers.logger")
+    def tests_generate_wealth_ethereum(self, mock_logging):
+        """Test generate wealth for ethereum."""
+        address = "my_address"
+        result = Response()
+        result.status_code = 500
+        with patch.object(requests, "get", return_value=result):
+            _try_generate_testnet_wealth(identifier=ETHEREUM, address=address)
+            assert mock_logging.error.called
+
+        result.status_code = 200
+        with pytest.raises(SystemExit):
+            with patch.object(requests, "get", return_value=result):
+                _try_generate_testnet_wealth(identifier=ETHEREUM, address=address)
```

Comment:
Since you are using the ResponseMock() function to test the other parts of the _try_generate_testnet_wealth, should be better to use the same thing here? ( I mean don't mind to delete existing code if you have a better solution).